axelor-production/src/main/java/com/axelor/apps/production/rest/ManufOrderProductRestServiceImpl.java

```diff
@@ -228,4 +242,73 @@ public void updateProdProductQty(ProdProduct prodProduct, BigDecimal qty) {
       prodProduct.setQty(qty);
     }
   }
+
+  /**
+   * Add consumed or produced product in manuf order.
+   *
+   * @param product
+   * @param qty
+   * @param trackingNumber
+   * @param manufOrder
+   * @param productType
+   * @return
+   */
+  @Transactional(rollbackOn = {Exception.class})
+  @Override
+  public StockMoveLine addManufOrderProduct(
+      Product product,
+      BigDecimal qty,
+      TrackingNumber trackingNumber,
+      ManufOrder manufOrder,
+      String productType)
+      throws AxelorException {
+
+    if (productType == null
+        || productType.isEmpty()
+        || (!PRODUCT_TYPE_CONSUMED.equals(productType)
+            && !PRODUCT_TYPE_PRODUCED.equals(productType))) {
+      throw new BadRequestException(
+          "Please, indicate the productType in the body : "
+              + PRODUCT_TYPE_CONSUMED
+              + " or "
+              + PRODUCT_TYPE_PRODUCED);
+    }
```

Comment:
This should be managed using RequestValidator.validateBody(requestBody);
with a annotation on productType